### PR TITLE
Validate the value of the arguments before sending the transaction

### PR DESCRIPTION
**Pull request by osipov-mit**
### Changes

- Rename `api.program.is` method to `api.program.exists` 
_(@nikitayutanov @Pinkdesu if it used somewhere pls update it)_
- Add validation of gasLimit, value and codeId before sending transactions
- `api.code.submit` method is now async 
_(@Pinkdesu pls update it on frontend)_
- Add getting some runtime constants (`existentialDeposit`, `blockGasLimit`, `mailboxTreshold`, `waitlistCost`)
- Add eslint config
---
### Bump `@gear-js/api` version to 0.23.4

### CODE REVIEW

File: api/src/Code.ts [hunk 1]
```diff
@@ -1,20 +1,30 @@
...
     return { codeHash, submitted: this.submitted };
   }
+
+  async exists(codeId: string) {
+    const codeMetadata = await this.api.query.gearProgram.metadataStorage(codeId) as Option<CodeMetadata>;
+    if (codeMetadata.isSome) {
+      return true;
+    }
+    return false;
+  }
```
Comment (Pinkdesu) [inline api/src/Code.ts:29]: return codeMetadada.isSome

File: api/src/Code.ts [hunk 2]
```diff
@@ -1,20 +1,30 @@
...
-    const codeHash = generateCodeHash(code);
     return { codeHash, submitted: this.submitted };
   }
+
+  async exists(codeId: string) {
+    const codeMetadata = await this.api.query.gearProgram.metadataStorage(codeId) as Option<CodeMetadata>;
+    if (codeMetadata.isSome) {
+      return true;
+    }
+    return false;
```
Comment (nikitayutanov) [inline api/src/Code.ts:28]:
```suggestion
    return codeMetadata.isSome ? true : false;
```

File: api/src/Program.ts [hunk 3]
```diff
@@ -76,7 +80,7 @@ export class GearProgram extends GearTransaction {
    * @param id some address in hex format
    * @returns if address belongs to program, method returns `true`, otherwise `false`
    */
-  async is(id: Hex): Promise<boolean> {
+  async exists(id: Hex): Promise<boolean> {
     const progs = await this.api.rpc.state.getKeys(GPROG);
     return progs.find((prog) => prog.eq(`0x${GPROG_HEX}${id.slice(2)}`)) ? true : false;
```
Comment (Pinkdesu) [inline api/src/Program.ts:85]: const program =  progs.find((prog) => prog.eq(`0x${GPROG_HEX}${id.slice(2)}`));

return Boolean(program);  // or !!program

File: api/src/utils/validate.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import { BN, u8aToBigInt } from '@polkadot/util';
+import { u128, u64 } from '@polkadot/types';
+
+import { GasLimit, Hex, Value } from '../types';
+import { ValidationError } from '../errors';
+import { GearApi } from '../GearApi';
+
+export function validateValue(value: Value | undefined, api: GearApi) {
+  if (value === undefined) return;
```
Comment (Pinkdesu) [inline api/src/utils/validate.ts:9]: !value

**Review by nikitayutanov — APPROVED** [review]

**Review by EugenWay — APPROVED** [review]

**Pull request merged**